purchase_fiscal_position_update/models/purchase.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# © 2011-2014 Julius Network Solutions SARL <contact@julius.fr>
+# @author Mathieu Vatel <mathieu _at_ julius.fr>
+# © 2014-2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.onchange('fiscal_position_id')
+    def fiscal_position_change(self):
```

Comment:
For me, it's a mistake to call onchange_product_id because it resets the price/description/... to the default value etc... And that's quite heavy too !